packages/auth/src/grant/routes.ts

```diff
@@ -114,13 +119,39 @@ async function createGrantInitiation(
   }
 }
 
+async function getGrantDetails(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  const secret = ctx.headers?.['x-idp-secret']
+  const { config, grantService } = deps
+  if (secret !== config.identityServerSecret) {
+    ctx.status = 401
+    return
+  }
+  const { id: interactId, nonce } = ctx.params
+  const grant = await grantService.getByInteractionSession(interactId, nonce)
+  if (!grant) {
+    ctx.status = 404
+    return
+  }
+
+  for (const access of grant.access) {
+    delete access.id
+    delete access.createdAt
+    delete access.updatedAt
+  }
+
+  ctx.body = { access: grant.access }
```

Comment:
We don't potentally leak senstive grant data to the idp do we?

Reply:
The scope of what is returned should only encompass the information the idp needs to present to the RO on its consent screen. Unless you would consider the recipient or amount of an `outgoing-payment` sensitive information I don't think there's anything in there we should obscure

Reply:
Basically, it could potentially return what's in the `access_token.access` field in this spec: https://docs.openpayments.guide/reference/post